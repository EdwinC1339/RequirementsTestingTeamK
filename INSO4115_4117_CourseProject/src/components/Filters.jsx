import React from "react";
import { useState, memo } from "react";
import Filter from "./Filter";
import Dropdown from "react-bootstrap/Dropdown";
import DropdownButton from "react-bootstrap/DropdownButton";

function Filters() {
  const restriction_names = [
    "Halal",
    "Lactose Intolerant",
    "Nut Allergy",
    "Vegan",
    "Vegetarian",
  ];

  const [selectedFilters, setSelectedFilters] = useState([]);

  const handleFilterSelection = (filterName) => {
    if (selectedFilters.includes(filterName)) {
      setSelectedFilters(selectedFilters.filter((name) => name !== filterName));
    } else {
      setSelectedFilters([...selectedFilters, filterName]);
    }
  };

  const handleCheckboxClick = (event) => {
    event.stopPropagation();
  };

  const filters = restriction_names.map((name) => {
    return (
      <Dropdown.Item key={name}>
        <label>
          <input
            type="checkbox"
            value={name}
            checked={selectedFilters.includes(name)}
            onChange={() => handleFilterSelection(name)}
            onClick={handleCheckboxClick}
          />
          {name}
        </label>
      </Dropdown.Item>
    );
  });

  return (
    <div className="filters-div">
      <h1>SELECT YOUR RESTRICTIONS</h1>
      <p>
        Select from an extensive list of dietary restrictions to be able to
        choose the food service that suits your needs
      </p>
      <DropdownButton
        variant="success"
        title="Restriction Selection"
        onToggle={() => {}}
        padding="10px"
      >
        {filters}
      </DropdownButton>
    </div>
  );
}

export default memo(Filters);
