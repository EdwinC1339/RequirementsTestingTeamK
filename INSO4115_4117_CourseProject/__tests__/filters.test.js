import React from "react";
import renderer from "react-test-renderer";
import Filters from "../src/components/Filters.jsx";

import { render } from "@testing-library/react";
import Filters from "./Filters";

test("Filters should render a dropdown list of items", () => {
  const { getByRole } = render(<Filters />);
  const dropdownButton = getByRole("button", { name: /Restriction Selection/ });
  expect(dropdownButton).toBeInTheDocument();
});
