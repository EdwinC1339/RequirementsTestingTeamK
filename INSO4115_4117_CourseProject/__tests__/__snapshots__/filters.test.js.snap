// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render a dropdown list of items 1`] = `
<div
  className="filters-div "
>
  <h1>
    SELECT YOUR RESTRICTIONS
  </h1>
  <p>
    Select from an extensive list of dietary restrictions to be able to choose the food service that suits your needs
  </p>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      className="dropdown-toggle btn btn-success"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      Restriction Selection
    </button>
  </div>
</div>
`;
